File: examples/mini/python/objects/gridlines.py
import goopylib as gp

window = gp.Window(812, 812, "Batch Rendering Lines!")

N = 100
distance = 812 / N

lines = []

for i in range(N):
    pos = distance * i - 406

    lines.append(gp.Line((pos, -406), (pos, 406), 2))
    lines.append(gp.Line((-406, pos), (406, pos), 2))

for line in lines:
    line.draw(window)


while window.is_open():
    gp.update()

gp.terminate()

File: examples/mini/cpp/objects/gridlines.cpp
#include "goopylib/goopylib.h"

int main() {
    gp::init();

    auto window = gp::Window(812, 812, "Batch Rendering Lines!");

    int N = 100;
    float distance = 812.0f / (float) N;

    std::vector<gp::Line> lines;

    for (int i = 0; i < N; i++) {
        float pos = distance * (float) i - 406;

        lines.push_back({{pos, -406.0f},
                         {pos, 406.0f}, 3.0f});

        lines.push_back({{-406.0f, pos},
                         {406.0f,  pos}, 3.0f});
    }

    for (auto line: lines) {
        line.draw(window);
    }

    while (window.isOpen()) {
        gp::update();
    }

    gp::terminate();
    return 0;
}
